autoemulate/experimental/data/utils.py

```diff
@@ -66,25 +68,29 @@ def _convert_to_dataloader(
                 f"Since x is already a DataLoader, expect y to be None, not {type(y)}."
             )
         else:
-            dataset = self._convert_to_dataset(x, y)
+            dataset = cls._convert_to_dataset(x, y)
             dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=shuffle)
         return dataloader
 
+    @classmethod
     def _convert_to_tensors(
-        self,
+        cls,
         x: InputLike,
         y: InputLike | None = None,
         dtype: torch.dtype = torch.float32,
     ) -> torch.Tensor | tuple[torch.Tensor, torch.Tensor]:
         """
         Convert InputLike x, y to Tensor or tuple of Tensors.
         """
-        dataset = self._convert_to_dataset(x, y)
+        dataset = cls._convert_to_dataset(x, y)
 
         # Handle Subset of TensorDataset
         if isinstance(dataset, Subset):
-            if isinstance(dataset.dataset, TensorDataset):
-                tensors = dataset.dataset.tensors
+            ds = dataset
+            while isinstance(ds, Subset):
+                ds = ds.dataset
+            if isinstance(ds, TensorDataset):
+                tensors = ds.tensors
```

Comment:
Added this as I think there were cases of nested `Subset` - does the indices look correct for this?

Reply:
Sorry missed the line in the quote: https://github.com/alan-turing-institute/autoemulate/pull/604/files#r2192232697